### PR TITLE
Alerting: Fix state cache id duplication

### PR DESCRIPTION
**Pull request by kylebrandt**
**What this PR does / why we need it**:
Makes state cache have a consistent key for entry:
 - Do not use string of map as key (as maps not ordered)
 - mix of including uid prefix and not, so not including since it ends up as __labels 

Otherwise entries duplicate as Grafana runs (should only be 2):
image

**Which issue(s) this PR fixes**:

Fixes https://github.com/grafana/alerting-squad/issues/117

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/services/ngalert/schedule/schedule.go [hunk 1]
```diff
@@ -355,10 +355,11 @@ func (sch *schedule) WarmStateCache(st *state.Manager) {
 		}
 		for _, entry := range cmd.Result {
 			lbs := map[string]string(entry.Labels)
+			cacheId, _, _ := entry.Labels.StringAndHash()
```
Comment (kylebrandt) [inline pkg/services/ngalert/schedule/schedule.go:358]: This is doing a sha1 has as well, so update to get just the string.

Comment (gotjosh) [reply pkg/services/ngalert/schedule/schedule.go]: Isn't it better to use the hash?

Comment (kylebrandt) [reply pkg/services/ngalert/schedule/schedule.go]: Could be, although not human readable in any way if debugging - and I already updated all those tests for the string :P

**Review by gotjosh — APPROVED** [review]
Fair enough. LGTM.

**Pull request merged**